collectors/node/metrics.go

```diff
@@ -15,225 +15,105 @@
-	return s
-}
 
-func getFilesystems() []nodeFilesystem {
-	f := []nodeFilesystem{}
-	parts, _ := disk.Partitions(false) // only physical partitions
-	for _, part := range parts {
-		usage, _ := disk.Usage(part.Mountpoint)
-		f = append(f, nodeFilesystem{
-			Name:               usage.Path,
-			CapacityTotalBytes: usage.Total,
-			CapacityUsedBytes:  usage.Used,
-			CapacityFreeBytes:  usage.Free,
-			InodesTotal:        usage.InodesTotal,
-			InodesUsed:         usage.InodesUsed,
-			InodesFree:         usage.InodesFree,
-		})
+	// For each metric poller defined, execute .poll() to get the latest
+	// metric, check for errors, and iterate over the slice of producers.Datapoint
+	// returned by .poll(), adding them to our top scope nodeMetrics slice.
+	for _, mp := range nodeMetricPollers {
+		if err := mp.poll(); err != nil {
+			return nc.datapoints, err
+		}
+
+		if err := mp.addDatapoints(&nc); err != nil {
+			return nc.datapoints, err
+		}
 	}
-	return f
-}
 
-func getNetworkInterfaces() []nodeNetworkInterface {
-	n := []nodeNetworkInterface{}
-	ioc, _ := net.IOCounters(true) // per nic
-	for _, nic := range ioc {
-		n = append(n, nodeNetworkInterface{
-			Name:      nic.Name,
-			RxBytes:   nic.BytesRecv,
-			TxBytes:   nic.BytesSent,
-			RxPackets: nic.PacketsRecv,
-			TxPackets: nic.PacketsSent,
-			RxDropped: nic.Dropin,
-			TxDropped: nic.Dropout,
-			RxErrors:  nic.Errin,
-			TxErrors:  nic.Errout,
-		})
-	}
-	return n
+	return nc.datapoints, nil
 }
 
-// calculatePct returns the percent utilization for CPU states. 100.00 => 100.00%
-func calculatePcts(lastTimes cpu.TimesStat, curTimes cpu.TimesStat) cpu.TimesStat {
-	totalDelta := curTimes.Total() - lastTimes.Total()
-	if totalDelta == 0 {
-		totalDelta = 1 // can't divide by zero
-	}
-	return cpu.TimesStat{
-		User:      round(math.Dim(curTimes.User, lastTimes.User) / totalDelta * 100),
-		System:    round(math.Dim(curTimes.System, lastTimes.System) / totalDelta * 100),
-		Idle:      round(math.Dim(curTimes.Idle, lastTimes.Idle) / totalDelta * 100),
-		Nice:      round(math.Dim(curTimes.Nice, lastTimes.Nice) / totalDelta * 100),
-		Iowait:    round(math.Dim(curTimes.Iowait, lastTimes.Iowait) / totalDelta * 100),
-		Irq:       round(math.Dim(curTimes.Irq, lastTimes.Irq) / totalDelta * 100),
-		Softirq:   round(math.Dim(curTimes.Softirq, lastTimes.Softirq) / totalDelta * 100),
-		Steal:     round(math.Dim(curTimes.Steal, lastTimes.Steal) / totalDelta * 100),
-		Guest:     round(math.Dim(curTimes.Guest, lastTimes.Guest) / totalDelta * 100),
-		GuestNice: round(math.Dim(curTimes.GuestNice, lastTimes.GuestNice) / totalDelta * 100),
-		Stolen:    round(math.Dim(curTimes.Stolen, lastTimes.Stolen) / totalDelta * 100),
-	}
-}
+// -- helpers
 
-// Helper function for rounding to two decimal places
-func round(f float64) float64 {
-	shift := math.Pow(10, float64(2))
-	return math.Floor(f*shift+.5) / shift
+func thisTime() string {
+	return time.Now().UTC().Format(time.RFC3339Nano)
```

Comment:
maybe it is a good idea to pass an argument to this function which will tell what format you want to use? and default to `time.RFC3339Nano` if none